kinde_sdk/kinde_api_client.py

```
@@ -338,16 +338,23 @@ def _decode_token_if_needed(self, token_name: str) -> None:
     def _decode_token_if_needed_value(self, token_name: str, token_value: dict) -> dict:
         token = token_value.get(token_name)
 
+      
+        if isinstance(token, bytes):
+            try:
+                token = token.decode('utf-8')
+            except UnicodeDecodeError:
+                raise KindeTokenException(f"Token {token_name} contains invalid UTF-8 bytes.")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)